File: common/src/main/java/generations/gg/generations/core/generationscore/common/network/packets/UnlockReloadPacketHandler.java
package generations.gg.generations.core.generationscore.common.network.packets;

import generations.gg.generations.core.generationscore.common.GenerationsDataProvider;
import generations.gg.generations.core.generationscore.common.network.ClientNetworkPacketHandler;
import net.minecraft.client.Minecraft;

public class UnlockReloadPacketHandler implements ClientNetworkPacketHandler<S2CUnlockReloadPacket> {
    public static final UnlockReloadPacketHandler INSTANCE = new UnlockReloadPacketHandler();

    public void handle(S2CUnlockReloadPacket packet, Minecraft minecraft) {
        GenerationsDataProvider.INSTANCE.canReload = true;
    }
}

File: common/src/main/java/generations/gg/generations/core/generationscore/common/network/packets/UnlockReloadPacketHandler.kt
package generations.gg.generations.core.generationscore.common.network.packets

import com.cobblemon.mod.common.api.net.ClientNetworkPacketHandler
import generations.gg.generations.core.generationscore.common.GenerationsDataProvider

import net.minecraft.client.Minecraft

object UnlockReloadPacketHandler : ClientNetworkPacketHandler<S2CUnlockReloadPacket> {
    override fun handle(packet: S2CUnlockReloadPacket, minecraft: Minecraft) {
        GenerationsDataProvider.INSTANCE.canReload = true
    }
}
